src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -407,6 +407,9 @@ public static void main(String args[]){
       if(args[0].equals("runByCategory")){
         boolean validCategoryName = false;
         for(ModuleID category : ModuleID.values()){
+          if(category!="UI"){
+            category = category.toLowerCase());
+          }
           if(category.name().equals(args[0]) && !(args[0].equals("TEST"))){
             validCategoryName = true;
```

Comment:
Something like category.name().toLowerCase().equals(args[0].toLowerCase()) && !(args[0].toLowerCase()).equals("test") works since it it is not altering any value.

Reply:
Can you make this change? We can do the final test and merge it.